### ISSUE TITLE
Deliverable for January 20th Site Map

### ISSUE THREAD

**Issue by thequbit** [mediapublic/mediapublic#78]
I know we've had a lot of discussion about this via email, and [here](https://github.com/mediapublic/mediapublic/wiki/Proposed-SIte-Layout).
- Home Page
  - Open for discussion on what should exist here
- Organization List
  - All the stations listed by state
  - Similar to how wikipedia has them [listed](https://en.wikipedia.org/wiki/List_of_NPR_stations)
  - Something _like_ how craigslist displays them: http://www.craigslist.org/about/sites )
- Organization Page
  - This is the station page ( @GabeIsman has this started based off of the design I did off of Melody's mock-up )
  - List of People
  - List of Podcasts
    - List of Recordings for that Podcast
- People Page?
  - I think this might need to be a stretch goal.
- Howto List
  - All the howto's, probably something like 10 at a time
    - Definitely paged
  - Search results here?
    - Maybe needs a different page?
    - Each item listed has:
      - Title
      - Description
      - Votes?
        - Maybe 1 - 5 star system?
      - Number of comments?
- Howto page
  - All the things to do with the howto
- Blog List
  - Standard issue list of blogs
  - Search?
    - Results here? or different page?
    - List of entries by month and all that jazz
- Blog
  - All the things to do with the blog post
- Bookmarklet
  - Bookmarklet to add podcasts and recordings ( not apart of the site, but needs to be written.  @ismcodes has been working on this )

I am going to start opening tickets to match this site outline and goals.

As always, open for discussion ( although I believe most of us signed off on this via email and/or previously when we went through it the first time ).

**Comment by GabeIsman** [comment]
:+1: This looks good to me. Don't think we need a separate page for the how to search results. The index should be flexible enough to display search results.

**Comment by thequbit** [comment]
After @ismcodes brought it up in #14, I realized I missed the entire "make a play list of recordings" feature set.

Do we still want to go after that for the MVP?  It sure would be slick, but I'm concerned that we could be coming close to not being able to deliver by January.

The proposed additional page(s) would be:
- Playlists
  - Would display a list of play lists the logged in user has
  - Would have a search bar that would let you search for other peoples play lists by name
- Playlist
  - A single playlist
  - Would support playing the recordings in the playlist
  - If you were the owner
    - You could delete recordings from a play list, and edit the name and description of a play list.
  - Maybe star ( upvote? ) a playlist?

I believe we already have the tables needed to support the many-to-many mapping of Recordings and Playlists in our ORM.

We would use the bookmarklet to add podcasts and recordings.  Then a user would add a recording to one of their playlists via the Organization page.  They would display their playlists from the `Playlists` page.  They would then play a play lists and manage ( edit ) a playlist from the `Playlist` page.

Thoughts from the group?

**Comment by nkkl** [comment]
How about we have a Playlists page, and when you're logged in you see a section of that this is for your playlists? Otherwise, you only see ones others have made.

We have to figure out the playlist curation flow, though - that's going to be a bunch of screens. This is a nontrivial amount of UX design work, and I'm going to prioritize the other stuff higher, so I don't know when I can promise this for. How much are you guys gated on design?

**Comment by thequbit** [comment]
I think with @gabeisman's edit in place framework, creating the new empty plastic is actually pretty simple. Then we'll just allow people to add recordings from the organization page with some kind of pull down. 

> How much are you guys gated on design?

We're not.  There is more than enough to do first.

**Comment by nkkl** [comment]
Sweet. You've been doing totally reasonable things with the UX so far, so I feel comfortable letting you pave this out. Definitely let me know as design questions come up and I'll help as I'm able.

**Comment by GabeIsman** [comment]
I'm concerned about adding playlists to the MVP. I think it adds a lot of stuff that's going to be kind of tricky to get right.

Are we going to have an audio player? That seems certainly outside of scope for MVP, and to me a lot of the value of a playlist, as a playlist consumer, is just hitting play and hearing a series of things. That will not be possible.

I don't know. To be honest I'm sort of concerned about our ability to deliver a quality MVP for January 20th WITHOUT the playlist stuff. Maybe I'm just worrying too much?

**Comment by thequbit** [comment]
@GabeIsman I agree.

I think we should take the Blog off the table for the MVP.  Drive hard to get the following done:
- Home Page ( #79 )
- Organizations Page ( #80 ) ( _view only + button/link for new_ )
  - Search bar
  - Stations by state and city ( as their call sign )
- Organization Page ( #81 ) ( _view, edit, new_ )
  - Station info
  - List of People ( which is actually Users )
  - List of Podcasts/Recordings
  - List of Help Requests ( opens an `mailto:` link )
- Howtos Page ( #83 ) ( _view only + button/link for new_ )
  - Search bar
  - Howtos list
    - Title
    - first 10 or so words of contents
- Howto page ( #84 ) ( _view, edit, new_ )
  - Title
  - Content
  - Organization Info
  - Author Info
- Help Request Page ( #114 ) ( _view, edit, new_ )
  - Title
  - Description
  - Expire Datetime
  - Organization Info
  - Author Info

_note: `Help Request` page is not designed yet by @nkkl because we just added it :/_

This means we do not do a playlist or a player.  If you want to listen to a recording, it takes you to the URL that was supplied by the bookmarklet, that's it.  If you want to listen to another recording, go back to mediapublic.io.

@melodykramer, @GabeIsman, @nkkl, @ismcodes, @ryansb  what do you think?

@GabeIsman I think you've got things set in motion here ( plus our convo last night ) that I can take on stubbing out anyone of the above pages.  I'll have some time on my hands this week where I can take a solid run at them if you're cool with that.  The `Organization` page is, by far, the most complicated/busy and arguably the most important.  If you want to keep owning that, I can take on the others with your insight/guidance.
